libs/src/sdk/api/albums/types.ts

```diff
@@ -0,0 +1,155 @@
+import { z } from 'zod'
+import type { CrossPlatformFile as File } from '../../types/File'
+import { Genre } from '../../types/Genre'
+import { HashId } from '../../types/HashId'
+import { Mood } from '../../types/Mood'
+import { isFileValid } from '../../utils/file'
+import { createUploadTrackMetadataSchema } from '../tracks/types'
+
+export const createUploadAlbumMetadataSchema = () =>
+  z
+    .object({
+      description: z.optional(z.string().max(1000)),
+      genre: z.enum(Object.values(Genre) as [Genre, ...Genre[]]),
+      isrc: z.optional(z.string()),
+      iswc: z.optional(z.string()),
+      license: z.optional(z.string()),
+      mood: z.optional(z.enum(Object.values(Mood) as [Mood, ...Mood[]])),
+      albumName: z.string(),
+      releaseDate: z.optional(
+        z.date().max(new Date(), { message: 'should not be in the future' })
+      ),
+      tags: z.optional(z.string()),
+      upc: z.optional(z.string())
+    })
+    .strict()
+
+export type AlbumMetadata = z.input<
+  ReturnType<typeof createUploadAlbumMetadataSchema>
+>
+
+const createAlbumTrackMetadataSchema = () =>
+  createUploadTrackMetadataSchema().partial({
+    genre: true,
+    mood: true,
+    tags: true
+  })
+
+export const createUploadAlbumSchema = () =>
+  z
+    .object({
+      userId: HashId,
+      coverArtFile: z.custom<File>((data: unknown) =>
+        isFileValid(data as File)
+      ),
+      metadata: createUploadAlbumMetadataSchema(),
+      onProgress: z.optional(z.function().args(z.number())),
+      /**
+       * Track metadata is populated from the album if fields are missing
+       */
+      trackMetadatas: z.array(createAlbumTrackMetadataSchema()),
+      trackFiles: z.array(
+        z.custom<File>((data: unknown) => isFileValid(data as File))
+      )
+    })
+    .strict()
+
+export type UploadAlbumRequest = z.input<
+  ReturnType<typeof createUploadAlbumSchema>
+>
+
+export const createUpdateAlbumSchema = () =>
+  z
+    .object({
+      userId: HashId,
+      albumId: HashId,
+      coverArtFile: z.optional(
+        z.custom<File>((data: unknown) => isFileValid(data as File))
+      ),
+      metadata: createUploadAlbumMetadataSchema()
+        .omit({ genre: true })
+        .merge(
+          z.object({
+            coverArtSizes: z.optional(z.string()),
+            isPrivate: z.optional(z.boolean())
+          })
+        ),
+      onProgress: z.optional(z.function().args(z.number()))
+    })
+    .strict()
+
+export type UpdateAlbumRequest = z.input<
+  ReturnType<typeof createUpdateAlbumSchema>
+>
+
+export const PublishAlbumSchema = z
```

Comment:
Since you can't create (upload) a private album (yet), prob doesn't make sense to have this right? cc @sliptype